pydicom/tests/test_valuerep.py

```diff
@@ -139,6 +139,14 @@ def testPickling(self):
         self.assertEqual(x.original_string,
                          x2.original_string)
 
+    def testLongInt(self):
+        # Check that a long int is read properly
+        # Will not work with enforce_valid_values
+        x = pydicom.valuerep.IS(3103050000)
+        data1_string = pickle.dumps(x)
+        x2 = pickle.loads(data1_string)
+        self.assertEqual(x.real, x2.real)
+
 
```

Comment:
Could you add another test that checks the correct exception? E.g. something like:
```python
    def testOverflow(self):
        original_flag = config.enforce_valid_values
        config.enforce_valid_values = True
        with pytest.raises(OverflowError, match="Value exceeds DICOM limits*"):
            pydicom.valuerep.IS(3103050000)      
        config.enforce_valid_values = original_flag
```